### PR TITLE
Harmonize test-suite with cookiecutter-pytask

### PR DESCRIPTION
**Pull request by timmens**
Harmonizes test-suite of repository with that of cookiecutter-pytask-project. This PR is not concerned with the tests found in {{cookiecutter.project_slag}}/tests. The latter problem will be fixed in a different PR.

I tried very hard to only mess with files that are relevant for the tests and the GitHub actions to run through, however, it turned out that this meant that I had to delete some files from the inner project and adjust some code and config files along the way.

### CODE REVIEW

**Comment by timmens** [comment]
The old code is really not compatible with the new code. To save time I will delete all old files, as we will be constructing a new example anyways.

**Comment by timmens** [comment]
To make the GitHub actions run through I had to 

- remove the interrogate hook from the inner project
- remove mypy from the inner project
- remove the HTML build test from ``test_check_conda_environment_creation_and_run_all_checks``

At least the first two points will be much easier to integrate once the new example is finished. I opened a new Issue so that we don't forget these points. I would propose finishing this PR first and then tackling the issue afterward, @hmgaudecker?

**Comment by hmgaudecker** [comment]
Thanks! 

Lest I missed it, just a small reminder:

> At least the first two points will be much easier to integrate once the new example is finished. I opened a new Issue so that we don't forget these points.

Can't see it?

File: cookiecutter.json [hunk 1]
```diff
@@ -33,6 +34,10 @@
     "yes",
     "no"
   ],
+  "use_biber_biblatex_for_tex_bibliographies": [
```
Comment (hmgaudecker) [inline cookiecutter.json:37]: I think we can just remove this. Should be stable enough to always use biber/biblatex.

**Review by hmgaudecker — APPROVED** [review]
Looks great, thanks!

**Comment by timmens** [comment]
I forgot to open the issue yesterday, thanks!

**Pull request merged**